From 2c69a58307f5234cbfee6cb377961c2c41bd8e01 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 11 Sep 2016 11:24:13 +0100
Subject: [PATCH 2/4] daemon/osd: Fix drunk OSD placement after movement

Ensure we always re-move the OSD prior to display lest it wander, as seen
this morning, where it moved one full unit east.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 daemon/osd.vala | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/daemon/osd.vala b/daemon/osd.vala
index 4942ed1..f0f4713 100644
--- a/daemon/osd.vala
+++ b/daemon/osd.vala
@@ -195,7 +195,7 @@ public class OSD : Gtk.Window
     /**
      * Move the OSD into the correct position
      */
-    private void move_osd()
+    public void move_osd()
     {
         /* Find the primary monitor bounds */
         Gdk.Screen sc = get_screen();
@@ -291,6 +291,9 @@ public class OSDManager
             Source.remove(expire_timeout);
             expire_timeout = 0;
         }
+        if (!osd_window.get_visible()) {
+            osd_window.move_osd();
+        }
         osd_window.show();
         expire_timeout = Timeout.add(timeout_length, this.osd_expire);
     }
-- 
2.10.0

